sklearn/cluster/tests/test_k_means.py

```diff
@@ -95,6 +118,28 @@ def test_elkan_results(distribution, tol):
     assert km_elkan.inertia_ == pytest.approx(km_full.inertia_, rel=1e-6)
 
 
+@pytest.mark.parametrize('distribution', ['normal', 'blobs'])
+def test_elkan_results_sparse(distribution):
+    # check that results are identical between lloyd and elkan algorithms
+    # with sparse input
+    rnd = np.random.RandomState(0)
+    if distribution == 'normal':
+        X = sp.random(100, 100, density=0.1, format='csr', random_state=rnd)
+        X.data = rnd.randn(len(X.data))
+    else:
+        X, _ = make_blobs(n_samples=100, n_features=100, random_state=rnd)
+        X = sp.csr_matrix(X)
+
+    km_full = KMeans(algorithm='full', n_clusters=5, random_state=0, n_init=1)
+    km_elkan = KMeans(algorithm='elkan', n_clusters=5,
+                      random_state=0, n_init=1)
+
+    km_full.fit(X)
+    km_elkan.fit(X)
+    assert_allclose(km_elkan.cluster_centers_, km_full.cluster_centers_)
+    assert_allclose(km_elkan.labels_, km_full.labels_)
```

Comment:
Maybe also check that this true for dense data X, or even better, that fitting on dense data gives the same results as fitting on the sparse data.

Reply:
`test_elkan_results` does it for the dense case. 
`test_full_vs_elkan` checks that the clustering is the same for both.